client/lib/config/test/index.js

```diff
@@ -50,5 +46,37 @@ describe( 'index', () => {
 
 			expect( result ).to.be.true;
 		} );
+
+		context( 'error cases', () => {
+			const NODE_ENV = process.env.NODE_ENV;
+			const fakeKey = 'where did all the errors go?';
+
+			afterEach( () => process.env.NODE_ENV = NODE_ENV );
+
+			it( `should throw an error when given key doesn't exist (NODE_ENV == development)`, () => {
+				process.env.NODE_ENV = 'development';
+
+				expect( () => config( fakeKey ) ).to.throw( ReferenceError );
+			} );
+
+			it( `should not throw an error when given key doesn't exist (NODE_ENV != development)`, () => {
+				const envs = [
+					'client',
+					'desktop',
+					'horizon',
+					'production',
+					'stage',
+					'test',
+					'wpcalypso',
+				];
+
+				envs.forEach( env => {
+					process.env.NODE_ENV = env;
```

Comment:
I think thats what Line 54 is doing: https://github.com/Automattic/wp-calypso/pull/11352/files/f54dcdaa74f59e8249b1af19feb875b4f2c31907..baae88d6c5bb627cbf1c06117eb4ad16feeaa51e#diff-2ed3a8b7ca52e886a50c4b8ee970509eR54